### ISSUE TITLE
debug_print flags print() in __main__ blocks

### ISSUE THREAD

**Issue by rsionnach** [rsionnach/sloppylint#8]
## Issue
The `debug_print` pattern flags legitimate `print()` calls in CLI applications and main blocks.

## Example
```python
if __name__ == "__main__":
    print("Starting application...")  # FLAGGED - but valid for CLI!
    main()
    print("Done!")  # FLAGGED
```

## Expected Behavior
`print()` statements in `if __name__ == "__main__":` blocks should not be flagged, as CLI applications legitimately use print for user output.

## Suggested Fix
- Skip print statements inside `if __name__ == "__main__":` blocks
- Consider also skipping in files named `cli.py`, `__main__.py`, or with CLI-related imports (click, argparse, typer)

**Comment by rsionnach** [comment]
Fixed in commit 703ca38. The `debug_print` pattern now:

1. **Skips CLI-related files**: `cli.py`, `__main__.py`, `main.py`, `console.py`, `commands.py`
2. **Skips CLI directories**: `scripts/`, `bin/`, `cli/`, `commands/`
3. **Skips files with CLI imports**: click, typer, argparse, fire, rich
4. **Skips `if __name__ == "__main__"` blocks**: Uses AST analysis to detect print() calls inside main blocks

This fix will be in version 0.3.0 on PyPI.
